templates/docs/examples/patterns/navigation/_script-sliding.js

```diff
@@ -106,30 +106,46 @@ const initNavigationSliding = () => {
   toggles.forEach(function (toggle) {
     toggle.addEventListener('click', function (e) {
       e.preventDefault();
+      closeSearch();
       const target = document.getElementById(toggle.getAttribute('aria-controls'));
       if (target) {
-        const isNested = target.parentNode.parentNode.classList.contains('p-navigation__dropdown');
-
+        const isNested = !target.closest('.p-navigation__dropdown');
         if (!isNested) {
           resetToggles(target);
         }
         if (target.getAttribute('aria-hidden') === 'true') {
           toggle.parentNode.classList.add('is-active');
           toggle.parentNode.parentNode.classList.add('is-active');
           target.setAttribute('aria-hidden', 'false');
+
+          // only animate the dropdown if menu is not open, otherwise just switch the visible one
+          if (!navigation.classList.contains('has-menu-open')) {
```

Comment:
It's based on how browser handles rendeing (and how the CSS transitions are triggered).

The `is-collapsed` class triggers a CSS transition. But, it if gets applied together with the rest of the state change that makes the element appear (display none is removed), the transition would not trigger (browser just paints it all together).

To trigger the CSS transition `is-collapsed` class name needs to be applied separately, in the next paint event. So that first, the elements get's rendered in "collapsed" state, and (a fraction of a milisecond later) the `is-collapsed` class will be applied and trigger CSS transition.

Don't know how to get it clearer from the code without an extensive comment.